frontend/device/wakeupmgr.lua

```diff
@@ -75,29 +73,61 @@ function WakeupMgr:addTask(seconds_from_now, callback)
 end
```

Comment:
You could resue `new_upcoming_task` in `self:setWakeupAlarm` (a few lines above):
```
    local new_upcoming_task = self._task_queue[1].epoch

    if not old_upcoming_task or (new_upcoming_task < old_upcoming_task) then
        self:setWakeupAlarm(new_upcoming_task)
    end
```

Reply:
Took me three days to grok what you meant, which should tell you how exhausted I was :D.

Done, thanks ;).

Reply:
In future, I will try to articulate myself clearer. But you could have asked. ;)

Anyway I enjoy the work here and with you (all).

Reply:
Oh, that was definitely a *me* thing, not to worry ;).

(Case in point: it wasn't even actually three days, it just felt like it ^^).